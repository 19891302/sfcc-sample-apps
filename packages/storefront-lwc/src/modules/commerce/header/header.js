/*
    Copyright (c) 2020, salesforce.com, inc.
    All rights reserved.
    SPDX-License-Identifier: BSD-3-Clause
    For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
*/
import { LightningElement } from 'lwc';

export default class CommerceHeader extends LightningElement {
    headerText = '';

    isHeaderText() {
        return this.headerText.length > 0;
    }

    connectedCallback() {
        window.addEventListener('setheadertext', event => {
            this.headerText = event.detail;
        });
    }
}
