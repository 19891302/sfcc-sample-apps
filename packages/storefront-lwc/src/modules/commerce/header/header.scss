@import "../../../scss/ux_variables";

.header {
  line-height: 3em;
}

.searchbar {
  padding: 0 1em .5em 1em;
  margin-top: .5em;
}

.sign-in span {
  margin: 0 .2em;
}

.mobile-menu {
  line-height: 3em;
  button {
    margin: 0;
    padding: 0;
    .menu-icon {
      margin-right: .2em;
    }
  }
}

.logo img {
  line-height: 3em;
  height: 2.5em;
}

.mini-basket {
  line-height: 3em;
}

.sub-header{
  height: 4.688em;
 
  @include mobile {
    height: 3.125em;
  }
}

.sub-header:before {
  content: '';
  position: absolute;
  top: 50%;
  left: 0;
  border-top: 0.063em solid $light-gray;
  background: $light-gray;
  width: 100%;
  transform: translateY(-50%);
}

.sub-header-content {
  background-color: white;
  width: auto;
  max-width: 90%;
  overflow: hidden;
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  color: $dark-gray;
  font-weight: bold;
  white-space: nowrap;
  border-left: 1em solid transparent;
  border-right: 1em solid transparent;
  text-overflow: ellipsis;

  @include mobile {
    font-size: 1.313em;
  }

  @include tablet {
    font-size: 1.5em;
  }

  @include desktop {
    font-size: 1.688em;
  }
  
  span.search-text {
    display: none;
 
    @include tablet {
      display: inline;
    }
    @include desktop {
      display: inline;
    }
  }
}
