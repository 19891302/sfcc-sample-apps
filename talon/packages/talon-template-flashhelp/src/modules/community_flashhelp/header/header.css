.navbar {
    min-height: 50px;
    margin-bottom: 20px;
    border: 1px solid transparent;
}

.navbar-header {
    display:flex;
    align-items: center;
}

.navbar-logo {
    margin-top: 4px;
}

.navbar-title {
    flex-grow: 1;
}

.navbar-default .navbar-brand:hover, .navbar-default .navbar-brand:focus {
    color: #5e5e5e;
    background-color: transparent;
}

.navbar-default {
    background-color: var(--lwc-header-background-color);
    border-color: #e7e7e7;
    font-size: 18px;
}

.navbar-default .navbar-brand {
   color: #777;
}

.navbar-logo {
    height: 50px;
    font-size: 18px;
    min-width: 170px;
    background-image: var(--lwc-company-logo);
    background-repeat:no-repeat;
    background-size:auto 90%;
    cursor: pointer;
}

.navbar-brand:hover, .navbar-brand:focus {
    text-decoration: none;
}

@media (min-width: 768px) {
    .navbar>.container .navbar-brand, .navbar>.container-fluid .navbar-brand {
        margin-left: -15px;
    }
}

.navbar-fixed-top {
    top: 0;
    border-width: 0 0 1px;
    position: fixed;
    right: 0;
    left: 0;
    z-index: 1030;
    transform: translate3d(0,0,0);
}

.icon-shopping-cart {
    text-align: center;
    font-size: 12px;
    width: 24px;
    height: 24px;
    line-height: 20.4px;
}

.icon-label {
    padding-left: 10px;
}

.container-fluid>.navbar-header {
    margin-right: 0;
    margin-left: 0;
}
