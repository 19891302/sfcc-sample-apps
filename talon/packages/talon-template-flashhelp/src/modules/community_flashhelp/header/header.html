<template>
    <div class="navbar navbar-default navbar-fixed-top">
        <div class="container-fluid">
            <div class="navbar-header">
                <div class="navbar-logo" onclick={handleLogoClick}>

                </div>
                <div class="navbar-title">
                    <span class="hidden-xs hidden-sm">Immediate Assistance Program</span>
                </div>
                <div class="navbar-right">
                    <community_flashhelp-shop-button></community_flashhelp-shop-button>
                </div>
            </div>
        </div>
    </div>
</template>
