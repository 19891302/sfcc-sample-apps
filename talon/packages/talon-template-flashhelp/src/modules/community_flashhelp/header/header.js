import { LightningElement } from 'lwc';
import { navigateToRoute } from "talon/routingService";

export default class Header extends LightningElement {
    handleLogoClick() {
        navigateToRoute('home');
    }
}
