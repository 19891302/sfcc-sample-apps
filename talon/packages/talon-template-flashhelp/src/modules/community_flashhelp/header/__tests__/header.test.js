import { createElement } from 'lwc';
import header from 'community_flashhelp/header';

describe('community_flashhelp/header', () => {
    beforeEach(() => {
        while (document.body.firstChild) {
            document.body.removeChild(document.body.firstChild);
        }
    });

    it('renders correctly', () => {
        const element = createElement('community_flashhelp-header', { is: header });
        document.body.appendChild(element);

        return Promise.resolve().then(() => {
            expect(element).toMatchSnapshot();
        });
    });
});
