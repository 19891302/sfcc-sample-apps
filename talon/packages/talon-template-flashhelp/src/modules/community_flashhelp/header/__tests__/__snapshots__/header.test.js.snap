// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`community_flashhelp/header renders correctly 1`] = `
<community_flashhelp-header>
  #shadow-root(open)
    <div
      class="navbar navbar-default navbar-fixed-top"
    >
      <div
        class="container-fluid"
      >
        <div
          class="navbar-header"
        >
          <div
            class="navbar-logo"
          />
          <div
            class="navbar-title"
          >
            <span
              class="hidden-xs hidden-sm"
            >
              Immediate Assistance Program
            </span>
          </div>
          <div
            class="navbar-right"
          >
            <community_flashhelp-shop-button>
              #shadow-root(open)
                <a
                  href="https://www.redcross.org/store"
                  style="text-decoration: none;"
                  target="_blank"
                >
                  <span
                    style="font-size: 14px; color: rgb(225, 27, 34); font-weight: bold; line-height: 24px;"
                  >
                    <i
                      class="fas fa-shopping-cart"
                    />
                    <span
                      class="icon-label"
                    >
                      Shop the Red Cross Store
                    </span>
                  </span>
                </a>
            </community_flashhelp-shop-button>
          </div>
        </div>
      </div>
    </div>
</community_flashhelp-header>
`;
